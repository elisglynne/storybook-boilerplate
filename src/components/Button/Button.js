import React, { Component } from 'react';
import styled from 'styled-components';

const StyledButton = styled.button`
  color: ${props => (props.primary ? '#fff' : '#4a4a4a')};
  background-color: ${props => (props.primary ? '#4a90e2' : 'transparent')};
  border: ${props => (props.primary ? 'none' : '2px solid #4a90e2')};
  border-radius: 8px;
  font-family: Helvetica, sans-serif;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: -0.45px;
  text-align: center;
  min-width: 200px;
  padding: 13.5px 30px;
  transition: 0.15s ease-in-out;

  &:hover {
    background-color: #3f7ec7;
  }

  &:active {
    transform: scale(0.9);
  }

  &:disabled {
    background-color: #9b9b9b;
    border: 2px solid #979797;
  }
`;

class Button extends Component {
  render() {
    return (
      <StyledButton
        onMouseDown={this.props.onClick}
        disabled={this.props.disabled}
        primary={this.props.primary}
      >
        {this.props.children}
      </StyledButton>
    );
  }
}

export default Button;
