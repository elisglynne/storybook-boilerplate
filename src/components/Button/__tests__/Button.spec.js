import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

it('renders <Button /> without throwing', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
