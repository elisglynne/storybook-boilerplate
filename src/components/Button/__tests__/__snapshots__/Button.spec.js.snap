// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Button /> without throwing 1`] = `
<button
  className="sc-bdVaJa eQzKrE"
/>
`;
